prometheus/registry.go

```diff
@@ -11,232 +11,253 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Copyright (c) 2013, The Prometheus Authors
-// All rights reserved.
-//
-// Use of this source code is governed by a BSD-style license that can be found
-// in the LICENSE file.
-
 package prometheus
 
 import (
-	"bytes"
-	"compress/gzip"
 	"errors"
 	"fmt"
-	"io"
-	"net/http"
-	"net/url"
 	"os"
 	"sort"
-	"strings"
 	"sync"
 
 	"github.com/golang/protobuf/proto"
-	"github.com/prometheus/common/expfmt"
+	"github.com/hashicorp/go-multierror"
```

Comment:
The problem is that something like multierror should be part of the standard library. Now that it is not, gazillions of implementations are flying around. This ones appeared to me to be the closest to a "canonical one". We could of course copy the implementation (bot note that it is not completely trivial and even pulls in yet another package).

This package would also be a candidate to library-internal vendoring.
(Although that would break people who want to type-convert the error.)

Internal vendoring should be considered for client_colang (but not in this PR).